sdk/src/quotes/public/swap-quote.ts

```diff
@@ -45,6 +46,56 @@ export type SwapQuote = {
   estimatedAmountOut: u64;
 } & SwapInput;
 
+export async function swapQuoteByInputToken(
+  whirlpool: Whirlpool,
+  swapTokenMint: Address,
+  tokenAmount: u64,
+  amountSpecifiedIsInput: boolean,
+  slippageTolerance: Percentage,
+  fetcher: AccountFetcher,
+  programId: Address,
+  refresh: boolean
+): Promise<SwapQuote> {
+  const whirlpoolData = whirlpool.getData();
+  const swapMintKey = AddressUtil.toPubKey(swapTokenMint);
+  invariant(
+    whirlpoolData.tokenMintA.equals(swapMintKey) || whirlpoolData.tokenMintB.equals(swapMintKey),
+    "swapTokenMint does not match any tokens on this pool"
+  );
+
+  const aToB =
+    swapMintKey.equals(whirlpoolData.tokenMintA) === amountSpecifiedIsInput ? true : false;
+
+  const tickArrayAddresses = PoolUtil.getTickArrayPublicKeysForSwap(
+    whirlpoolData.tickCurrentIndex,
+    whirlpoolData.tickSpacing,
+    aToB,
+    AddressUtil.toPubKey(programId),
+    whirlpool.getAddress()
+  );
+  const tickArrays = await fetcher.listTickArrays(tickArrayAddresses, refresh);
+
+  // Check if all the tick arrays have been initialized.
+  const uninitializedIndices = TickArrayUtil.getUninitializedArrays(tickArrays);
+  if (uninitializedIndices.length > 0) {
+    const uninitializedArrays = uninitializedIndices.reduce<string>(
```

Comment:
nit: I think it could be cleaner to do a map then join?

```
unitializedIndices.map(index => tickArrayAddresses[unInitIndex].toBase58()).join(" ");
```

Reply:
you learn something new everyday!